core/field_variable.js

```diff
@@ -129,6 +132,31 @@ Blockly.FieldVariable.prototype.setValue = function(value) {
 };
 
 /**
+ * Return a list of variable types to include in the dropdown.
+ * @return {!Array.<string>} Array of variable types.
+ * @throws {Error} if variableTypes is an empty array.
+ * @private
+ */
+Blockly.FieldVariable.prototype.getVariableTypes_ = function() {
+  var variableTypes = this.variableTypes;
+  if (variableTypes === null) {
```

Comment:
Woops, I just double checked. If it is explicitly null it is all types. It's the undefined case that it defaults to ''